extensions/amp-access/0.1/amp-access.js

```diff
@@ -814,10 +817,11 @@ export class AccessService {
    *
    * @param {string} url
    * @param {string} eventLabel A label used for the analytics event for this action
+   * @param {boolean} noReturnUrl Don't perform a RETURN_URL replacement
    * @return {!Promise}
    */
-  loginWithUrl(url, eventLabel = '') {
-    return this.login_(url, eventLabel);
+  loginWithUrl(url, eventLabel = '', noReturnUrl) {
```

Comment:
I see. Here's how this now works. Let's see if it's possible at all to reconcile.

1. We try to open login dialog ourselves as a popup. Return URL points to the ["login-done" page](https://github.com/ampproject/amphtml/blob/95b5a08122279915b1beb9bd30d9048d2ce6eb9a/extensions/amp-access/0.1/login-dialog.js#L289) that messages back to the parent page to refresh.
2. (1) is an ideal scenario, but, in many cases, that doesn't work - the browser cannot or would not open a popup. E.g. a WebView or CCT. Instead, the browser would simply top-navigate to the login dialog. In this case, our "login-done" page further redirects back to the original page.
3. We allow a parent web or native app to assist to show the login dialog. This is somewhat common for a webview. What happens in this case, we message back to the parent app with `loginUrl?return=RETURN_URL`. The parent app substitutes `RETURN_URL` to whatever it can work with. It'd often be something like `my-native-app-protocol:some-return-code`. Then everything works as before and the native app then messages back success/error code to the AMP runtime.

All three cases are completely equivalent from the point of view of the login dialog. But obviously, given the "signing" needs on your side that's not the case.

I see two ways forward:

a. You could whitelist `?return=RETURN_URL` query parameter to be excluded from signing. E.g. we could *always* force it at the end of the URL and you'd chop it off before verifying signature. That would, of course, work if you don't see any security issue with the return redirect, which in turn would normally depend on what you pass back. But if you only pass back `#success` or such - probably not a security problem?
b. So far, I believe 1-2-3 above are simply variations on trying to do best effort to show login dialog better. We could force the login for you into options 1 or 2. But I need to check this. This might have been added for cases where launching login dialog via parent app was required for some purpose, other than just "enhancement".

What do you think about the option (a)?